src/nupic/algorithms/Connections.cpp

```diff
@@ -418,78 +389,35 @@ bool segmentIncreasingOrder(const SegmentOverlap& a, const SegmentOverlap& b)
   return a.segment < b.segment;
 }
 
-void Connections::computeActivity(const vector<Cell>& input,
+void Connections::computeActivity(const vector<CellIdx>& input,
                                   Permanence activePermanenceThreshold,
                                   SynapseIdx activeSynapseThreshold,
                                   Permanence matchingPermanenceThreshold,
                                   SynapseIdx matchingSynapseThreshold,
-                                  vector<SegmentOverlap>& outActiveSegments,
-                                  vector<SegmentOverlap>& outMatchingSegments,
-                                  bool recordIteration)
+                                  vector<SegmentOverlap>& activeSegmentsOut,
+                                  vector<SegmentOverlap>& matchingSegmentsOut)
+  const
 {
-  vector<UInt32> numActiveSynapsesForSegment(nextFlatIdx_, 0);
-  vector<UInt32> numMatchingSynapsesForSegment(nextFlatIdx_, 0);
+  SegmentExcitationTally excitations(*this, activePermanenceThreshold,
```

Comment:
Your links aren't working and I can't figure out what you mean. Can you elaborate?